Xamarin.Forms.Platform.UAP/Platform.cs

```diff
@@ -345,19 +345,23 @@ async void SetCurrent(Page newPage, bool popping = false, bool modal = false, Ac
 					Page previousPage = _currentPage;
 
 					if (modal && !popping && !newPage.BackgroundColor.IsDefault)
-						_modalBackgroundPage = previousPage;
```

Comment:
Because of this line, when you navigate to the second modal level, `_modalBackgroundPage` already has a value and will be overwritten.
As of that, when you navigate back the second time, `_modalBackgroundPage` will already be `null` (this is done in the first pop of the navigation stack at line 360 below!). So you lost the outer modal page and cannot call its `OnAppeare()` callback anymore!